pc/registration/icp/icp.go

```diff
@@ -0,0 +1,96 @@
+package icp
+
+import (
+	"errors"
+	"time"
+
+	"github.com/seqsense/pcgol/mat"
+	"github.com/seqsense/pcgol/pc"
+	"github.com/seqsense/pcgol/pc/storage"
+
+	"github.com/seqsense/pcgol/internal/gnuplot"
+)
+
+var (
+	ErrNeedGradient = errors.New("need gradient output of Evaluator")
+)
+
+var DefaultGradientWeight = mat.Vec6{0.3, 0.3, 0.3, 0.3, 0.3, 0.3}
```

Comment:
My understanding of the weights was that it is a way to "bias" the transformation towards some dimensions but if they are all the same, it is pretty much a scaling factor. As the delta transformation is already scaled by `factorIter` at each iteration, why not defaulting to `1`?

Reply:
When I set it to 1, it often overshoots and not converged well.
In gradient descent method, factor should be just `small enough`. https://en.wikipedia.org/wiki/Gradient_descent